### PR TITLE
Feat/colab cli 3 train

### PR DESCRIPTION
**Pull request by XiandaDu**
# Pull Request Checklist

## Reference Issue

ref: N/A

## Summary

- Adds `colab/train_on_colab.sh`: a one-command recipe that provisions a Colab GPU, runs the existing full-parameter accelerate/DeepSpeed fine-tuning flow (`training/train_rankllm.py`) remotely on a single GPU, and downloads the checkpoint as a `.tar.gz`.
- Adds `training/configs/accel_config_single_gpu.yaml` (DeepSpeed ZeRO-3, single process) used by the recipe.
- Fully env-var configurable (GPU, base model, objective, epochs, batch/accumulation, `EXTRA_TRAIN_ARGS`, …); `DRY_RUN=1` supported.
- Completes `docs/colab.md` and `colab/README.md` with training usage and GPU sizing (full-parameter 7B needs A100-80GB/H100; no LoRA path exists), and adds a pointer in the root README Training section.

## Why

PR **3/3** of the Colab CLI recipes stack, stacked on **#\<PR2\>** (`feat/colab-cli-2-rerank`). Reuses the shared bootstrap from 1/3. With all three merged, the tree matches the original tested `feat/colab-cli-recipes` branch exactly. Lets users reproduce RankZephyr/FirstMistral-style fine-tuning without local GPU hardware.

## Validation

```bash
bash -n colab/train_on_colab.sh
DRY_RUN=1 bash colab/train_on_colab.sh
GPU=H100 OBJECTIVE=combined EXTRA_TRAIN_ARGS="--ranking_loss ranknet --weighted" DRY_RUN=1 bash colab/train_on_colab.sh
.venv/bin/pre-commit run --all-files
```

Verified the dry-run command sequence (provision → warmup → exec → download → stop) and the injected remote `CONFIG` for both default and override cases.

## Follow-ups

- None planned; possible later: multi-GPU config variant and a CI-run offline smoke check for the recipes.

## Checklist Items

Before submitting your pull request, please review these items:

- [x] Have you followed the [contributing guidelines](CONTRIBUTING.md)?
- [x] Have you verified that there are no existing [Pull Requests](https://github.com/castorini/rank_llm/pulls) for the same update/change?
- [x] Have you updated any relevant documentation or added new tests where needed?

# PR Type

What kind of change does this PR introduce?

- [ ] Bugfix
- [x] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no API changes)
- [ ] Documentation content changes
- [ ] Reproduction logs
- [ ] Other...
    - Description:

### CODE REVIEW

File: colab/train_on_colab.sh [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+WORKDIR="${WORKDIR:-/content/rank_llm}"
+BASE_MODEL="${BASE_MODEL:-HuggingFaceH4/zephyr-7b-beta}"
+TRAIN_DATA_PATH="${TRAIN_DATA_PATH:-rryisthebest/rank_zephyr_training_data_alpha}"
+OBJECTIVE="${OBJECTIVE:-generation}"
+OUTPUT_DIR="${OUTPUT_DIR:-$WORKDIR/training/models/colab_run}"
+NUM_TRAIN_EPOCHS="${NUM_TRAIN_EPOCHS:-1}"
+PER_DEVICE_TRAIN_BATCH_SIZE="${PER_DEVICE_TRAIN_BATCH_SIZE:-1}"
+GRADIENT_ACCUMULATION_STEPS="${GRADIENT_ACCUMULATION_STEPS:-16}"
+NUM_WARMUP_STEPS="${NUM_WARMUP_STEPS:-10}"
+INSTALL_FLASH_ATTN="${INSTALL_FLASH_ATTN:-0}"
```
[the remark this answers is not in the document]
Comment (XiandaDu) [reply colab/train_on_colab.sh]: This makes sense. I believed that loading flash attention in a temporary environment is too expensive, but our previous implementation in model_utils.py loaded flash attention by default. Thus. I changed toload flash attention in Colab here as well, but note that this may increase the environment build time on each run.
